comp/dogstatsd/mapper/mapper.go

```diff
@@ -25,6 +23,29 @@ const (
 	matchTypeRegex    = "regex"
 )
 
+//
+// Those two struct are used to pull data from the configuration into typed struct. We currently load the data from the
+// configurarion into MappingProfileConfig and then convert it to MappingProfile.
+//
+// Both MappingProfileConfig and MetricMappingConfig will be removed once we remove `UnmarshalKey` method from the
```

Comment:
How about updating this to

```suggestion
// Both MappingProfileConfig and MetricMappingConfig will no longer use `mapstructure` once we remove `UnmarshalKey` method from the config interface
```